sympy/geometry/polygon.py

```diff
@@ -783,6 +783,87 @@ def intersection(self, o):
             return list(ordered(intersection_result))
 
 
+    def cut_section(self, line):
+        """
+        Returns a tuple of two polygon segments that lie above and below
+        the intersecting line respectively.
+
+        References
+        ==========
+
+        https://github.com/sympy/sympy/wiki/A-method-to-return-a-cut-section-of-any-polygon-geometry
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Symbol, Polygon, Line
+        >>> a, b = 20, 10
+        >>> p1, p2, p3, p4 = [(0, b), (0, 0), (a, 0), (a, b)]
+        >>> rectangle = Polygon(p1, p2, p3, p4)
+        >>> t = rectangle.cut_section(Line((0, 5), slope=0))
+        >>> t
+        (Polygon(Point2D(0, 10), Point2D(0, 5), Point2D(20, 5), Point2D(20, 10)),
+        Polygon(Point2D(0, 5), Point2D(0, 0), Point2D(20, 0), Point2D(20, 5)))
+        >>> upper_segment, lower_segment = t
+        >>> upper_segment.area
+        100
+        >>> upper_segment.centroid
+        Point2D(10, 15/2)
+        >>> lower_segment.centroid
+        Point2D(10, 5/2)
+        """
+        intersection_points = self.intersection(line)
+        if not intersection_points:
+            raise ValueError("This line does not intersect the polygon")
+
+        points = self.vertices
+        points.append(points[0])
+
+        x, y = symbols('x, y', real=True, cls=Dummy)
+        eq = line.equation(x, y)
+
+        # considering equation of line to be `ax +by + c`
+        a = eq.coeff(x)
+        b = eq.coeff(y)
+
+        upper_vertices = []
+        lower_vertices = []
+        # prev is true when previous point is above the line
+        prev = True
+        prev_point = None
+        for point in points:
+            # when coefficient of y is 0, right side of the line is
+            # considered
+            compare = (eq.evalf(subs={x: point.x, y: point.y}))/b if b \
+                    else (eq.subs(x, point.x))/a
```

Comment:
Can this not just be `eq.subs({x: point.x, y:point.y})`?

Reply:
In general, using `evalf(subs...)` is better because it allows evalf to increase precision on the given numbers to get the desired precision when cancellation might otherwise give a low precision result. I imagine this might be an issue when a line cuts a polygon a distance less than 1e-16 (or so) from a vertex.

Reply:
My point was really: is evalf needed here?

Reply:
I hadn't thought of `evalf()` in such a way, but initially, my tests were failing with `subs()` being used so I tried it the other way. I agree there is no need of `evalf()` here. So, I have changed it to `.subs()` it works fine.